docs/tutorials/semantic_search/generate_embeddings_for_arrays_of_object.md

```diff
@@ -0,0 +1,299 @@
+- remove processor: remove temporary field
+```
+PUT _ingest/pipeline/bedrock_embedding_pipeline
+{
+  "processors": [
+    {
+      "set": {
+        "field": "title_tmp",
+        "value": "{{_ingest._value.title}}"
+      }
+    },
+    {
+      "text_embedding": {
+        "model_id": your_embedding_model_id,
+        "field_map": {
+          "title_tmp": "_ingest._value.title_embedding"
+        }
+      }
+    },
+    {
+      "remove": {
+        "field": "title_tmp"
+      }
+    }
+  ]
+}
+```
+
+Create pipeline with foreach processor:
+```
+PUT _ingest/pipeline/bedrock_embedding_foreach_pipeline
+{
+  "description": "Test nested embeddings",
+  "processors": [
+    {
+      "foreach": {
+        "field": "books",
+        "processor": {
+          "pipeline": {
+            "name": "bedrock_embedding_pipeline"
+          }
+        },
+        "ignore_failure": true
+      }
+    }
+  ]
+}
+```
+
+### 2.3 Simulate pipeline
+
+- Case1: two book objects with title
+```
+POST _ingest/pipeline/bedrock_embedding_foreach_pipeline/_simulate
+{
+  "docs": [
+    {
+      "_index": "my_books",
+      "_id": "1",
+      "_source": {
+        "books": [
+          {
+            "title": "first book",
+            "description": "This is first book"
+          },
+          {
+            "title": "second book",
+            "description": "This is second book"
+          }
+        ]
+      }
+    }
+  ]
+}
+```
+Response
+```
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "my_books",
+        "_id": "1",
+        "_source": {
+          "books": [
+            {
+              "title": "first book",
+              "title_embedding": [-1.1015625, 0.65234375, 0.7578125, ...],
+              "description": "This is first book"
+            },
+            {
+              "title": "second book",
+              "title_embedding": [-0.65234375, 0.21679688, 0.7265625, ...],
+              "description": "This is second book"
+            }
+          ]
+        },
+        "_ingest": {
+          "_value": null,
+          "timestamp": "2024-05-28T16:16:50.538929413Z"
+        }
+      }
+    }
+  ]
+}
+```
+- Case2: book object without title
+```
+POST _ingest/pipeline/bedrock_embedding_foreach_pipeline/_simulate
+{
+  "docs": [
+    {
+      "_index": "my_books",
+      "_id": "1",
+      "_source": {
+        "books": [
+          {
+            "title": "first book",
+            "description": "This is first book"
+          },
+          {
+            "description": "This is second book"
+          }
+        ]
+      }
+    }
+  ]
+}
+```
+Response
+```
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "my_books",
+        "_id": "1",
+        "_source": {
+          "books": [
+            {
+              "title": "first book",
+              "title_embedding": [-1.1015625, 0.65234375, 0.7578125, ...],
+              "description": "This is first book"
+            },
+            {
+              "title": "second book",
```

Comment:
how did you get this `title` here?

Reply:
Good catch, should remove this, will fix in next PR soon